### PR TITLE
And one more fix to flaky tests

### LINKED ISSUE #5144: Flaky transaction pool test
```
 ---- testing::pool::should_push_watchers_during_maintaince stdout ----
thread 'testing::pool::should_push_watchers_during_maintaince' panicked at 'assertion failed: `(left == right)`
  left: `5`,
 right: `3`', client/transaction-pool/src/testing/pool.rs:331:2
```
Job: https://gitlab.parity.io/parity/substrate/-/jobs/407602

cc @tomusdrw @NikVolf

### PR DESCRIPTION
**Pull request by NikVolf**
(testing timers is hard, but it should nail it)

Closes #5144

### CODE REVIEW

File: client/transaction-pool/src/testing/pool.rs [hunk 1]
```diff
@@ -219,10 +219,9 @@ fn should_revalidate_during_maintenance() {
 
 	pool.api.push_block(1, vec![xt1.clone()]);
 
-	notifier.clear();
 	block_on(pool.maintain(block_event(1)));
 	assert_eq!(pool.status().ready, 1);
-	block_on(notifier.next());
+	block_on(notifier.next_blocking());
```
Comment (bkchr) [inline client/transaction-pool/src/testing/pool.rs]: What is the difference between next and next_blocking?

Comment (NikVolf) [reply client/transaction-pool/src/testing/pool.rs]: `next_blocking` first clears the queue from all previous timer events, then starts to wait next one (same as `clear()` + `next()` before)

File: Cargo.lock [hunk 2]
```diff
@@ -2347,9 +2347,9 @@ checksum = "141340095b15ed7491bd3d4ced9d20cebfb826174b6bb03386381f62b01e3d77"
 
 [[package]]
 name = "intervalier"
-version = "0.3.0"
+version = "0.3.1"
```
Comment (bkchr) [inline Cargo.lock]: This should be required in the `Cargo.toml`

File: client/transaction-pool/src/testing/pool.rs (hunk 1 above)
Comment (bkchr) [reply client/transaction-pool/src/testing/pool.rs]: Hmmm. If that is the case, what did change to the version before than? I don't see any difference.

Comment (NikVolf) [reply client/transaction-pool/src/testing/pool.rs]: Now it clears events after `maintain`, so if during `maintain` the timer is fired, it is not handled before maintain is finished 😅

File: client/transaction-pool/src/testing/pool.rs [hunk 3]
```diff
@@ -263,14 +262,10 @@ fn should_not_retain_invalid_hashes_from_retracted() {
 
 	let event = block_event_with_retracted(1, vec![retracted_hash]);
 
-	notifier.clear();
 	block_on(pool.maintain(event));
 	// maintenance is in background
-	block_on(notifier.next());
-
-	let event = block_event_with_retracted(1, vec![retracted_hash]);
```
Comment (tomusdrw) [inline client/transaction-pool/src/testing/pool.rs]: We lost one test case it seems?

File: client/transaction-pool/src/testing/pool.rs (hunk 1 above)
Comment (bkchr) [reply client/transaction-pool/src/testing/pool.rs]: Ahh, so we just than wait one iteration more?

File: client/transaction-pool/src/testing/pool.rs (hunk 3 above)
Comment (NikVolf) [reply client/transaction-pool/src/testing/pool.rs]: it was duplicated code that didn't add anything

File: client/transaction-pool/src/testing/pool.rs (hunk 1 above)
Comment (NikVolf) [reply client/transaction-pool/src/testing/pool.rs]: yes, one iteration that is guaranteed to be handled by timer handler

**Pull request merged**